### PR TITLE
Documentation and some fixes along the way

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
These are some updates for a 0.6.0 release.  Planning on finishing up documentation in another PR.

- Update changelog
- [`Breaking`] Change type from Transaction to CommittedTransaction for cases that it's known they're committed
- [`Fix`] Fix some of the API types that didn't match on-chain representations
- Add Go doc for most functions and types in the codebase
- Add tons more testing
- [`Breaking`] Change ToAnyPublicKey to have an error in the output

### Test Plan
See unit tests

### CODE REVIEW

File: api/blocks_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	assert.Equal(t, "0x014e30aafd9f715ab6262322bf919abebd66d948f6822ffb8a2699a57722fb80", data.BlockHash)
+	assert.Equal(t, uint64(1665609760857472), data.BlockTimestamp)
+	assert.Equal(t, uint64(1), data.BlockHeight)
+	assert.Equal(t, uint64(1), data.FirstVersion)
+	assert.Equal(t, uint64(1), data.LastVersion)
+	assert.Empty(t, data.Transactions)
+}
+
+func TestBlockWithTransactions(t *testing.T) {
```
Comment (brianolson) [inline api/blocks_test.go:51]: copy-paste-o? looks the same as 'with NO transactions'

File: internal/types/account.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// Package types is an internal package that contains the types used in the Aptos Go SDK.
```
Comment (brianolson) [inline internal/types/account.go:1]: I think the convention is for this kind of note to go in `doc.go` (with no other code content)

File: api/blocks_test.go (hunk 1 above)
Comment (gregnazario) [reply api/blocks_test.go:51]: Yeah I think my PR got too big, and I just cut it off.  Just have missed this

File: internal/types/account.go (hunk 2 above)
Comment (gregnazario) [reply internal/types/account.go:1]: All the doc guides, don't give enough conventions, just worry about syntax.  Will look into this

**Pull request merged**